src/verify.rs

```diff
@@ -446,13 +420,10 @@ pub struct VerifyResult {
 
 impl VerifyResult {
     pub fn diagnostics(&self) -> Vec<(StatementAddress, Diagnostic)> {
-        let mut out = Vec::new();
-        for vsr in self.segments.values() {
-            for (&sa, &ref diag) in &vsr.diagnostics {
-                out.push((sa, diag.clone()));
-            }
-        }
-        out
+        self.segments
+            .values()
+            .flat_map(|vsr| vsr.diagnostics.iter().map(|(&sa, diag)| (sa, diag.clone())))
```

Comment:
I would say that it is more readable, but not easier to debug - that is, there a lot of other incorrect ways to write it that also look reasonable but are wrong for inscrutable reasons. If it works, though, I think it gives a nice declarative description of the return value. The other reasons I like this approach is (1) I love iterators and that Rust has them and (2) I have an urge to minimize code the same as minimizing set.mm proofs.